adoc/extensions/sycl_khr_command_graph.adoc

```diff
@@ -252,111 +253,423 @@ a@ Constructs a [code]#depends_on_all_leaves# property instance.
+[code]#queue_state::executing# state.
+
+_Effects (2):_ Synchronously changes the state of [code]#recordingQueue# to the
+[code]#queue_state::executing# state.
+This operation is a no-op if [code]#recordingQueue# is already in the
+[code]#queue_state::executing# state.
+
+_Effects (3):_ Synchronously changes the state of each queue in
+[code]#recordingQueues# to the [code]#queue_state::executing# state.
+This operation is a no-op for any queue in [code]#recordingQueues# that is
+already in the [code]#queue_state::executing# state.
+
+_Throws (2)(3):_
+
+* An [code]#exception# with the [code]#errc::invalid# error code if any queue is
+recording to a different graph.
+
+.[apititle]#khr::command_graph::add (empty node)#
+[source, role=synopsis,id=api:khr-command-graph-add-empty]
+----
+node add(const property_list& propList = {})
+----
+_Constraints:_ [code]#State# is [code]#graph_state::modifiable#.
+
+_Effects:_ This creates an empty node which contains no command.
+Empty nodes can be used to create transitive dependencies between other graph
+nodes, similar to a barrier.
+Zero or more properties can be provided to this function via an instance of
+[code]#property_list#.
+Valid properties for this function are defined in
+<<subsec:khr-graph-node-properties>>.
+
+_Returns:_ The empty node which has been added to the graph.
+
+_Throws:_ An [code]#exception# with the [code]#errc::invalid# error code if a
+queue is recording commands to the graph.
+
+.[apititle]#khr::command_graph::add#
+[source, role=synopsis,id=api:khr-command-graph-add]
+----
+template <typename T>
+  node add(T cgf, const property_list& propList = {})
+----
+_Constraints:_ [code]#State# is [code]#graph_state::modifiable#.
+
+_Effects:_ The [code]#cgf# command group function behaves in much the same way
+as the command group function passed to [code]#queue::submit# unless explicitly
+stated otherwise.
+Code in the function is evaluated synchronously, before the function returns
+back to [code]#command_graph::add#, with the exception of any SYCL commands
+(e.g. kernels, or explicit memory copy operations).
+These commands are captured into the graph and executed asynchronously when the
+graph is submitted to a queue.
+The requisites of [code]#cgf# will be used to identify any dependent nodes in
+the graph to form edges with.
+Zero or more properties can be provided to this function via an instance of
+[code]#property_list#.
+Valid properties for this function are defined in
+<<subsec:khr-graph-node-properties>>.
+
+
+_Returns:_ The node which has been added to the graph.
+
+_Throws:_
+
+* An [code]#exception# with the [code]#errc::invalid# error code if a queue is
+  recording commands to the graph.
+* An [code]#exception# with the [code]#errc::invalid# error code if the graph
+  wasn't created with the
+  [code]#property::command_graph::assume_buffer_outlives_graph# property and
+  this command uses a buffer.
```

Comment:
I'd maybe rephrase this as the command using an accessor rather than a buffer, as the buffer isn't used directly within the command group.